diff --git a/node_modules/hexo-filename-date/index.js b/node_modules/hexo-filename-date/index.js
index b8aba82..58caf07 100644
--- a/node_modules/hexo-filename-date/index.js
+++ b/node_modules/hexo-filename-date/index.js
@@ -1,7 +1,7 @@
 const path   = require('path')
     , moment = require('moment')
 
-const RE_FILENAME = /^(\d{4}-\d{2}-\d{2})-(.*)$/
+const RE_FILENAME = /^(\d{4}-\d{1,2}-\d{1,2})-(.*)$/
 
 hexo.extend.filter.register('before_post_render', data => {
   if (data.layout === 'post') {
@@ -9,7 +9,7 @@ hexo.extend.filter.register('before_post_render', data => {
         , matches  = filename.match(RE_FILENAME)
 
     if (matches) {
-      data.date || (data.date = moment(matches[1]))
+      data.date = moment(matches[1]);
       data.title || (data.title = matches[2])
     }
   }
